schematools/importer/base.py

```diff
@@ -478,19 +478,54 @@ def index_factory(
     In case of temporal data, this will lead to intersection tables a.k.a. through tables to
     accomodate n:m relations.
 
+    FK index:
+    In the JSON schema definition of the table, relations may be 1:N relations definied.
+    Where the child table is referencing a parent table.
+    These reference are not enforced by a database foreign key constraint. But will be
+    used in joins to collect data when calling a API endpoint. Therefore must be indexed
+    for optimal performance.
+
     The returned Index objects are keyed on the name of table
     """
 
     index = dict()
     metadata = metadata or MetaData()
     logger = LogfileLogger(logger) if logger else CliLogger()
+    table_object = metadata.tables[dataset_table.db_name()]
+    indexes_to_create = []
+
+    def make_hash_value(index_name):
+        """
+        Postgres DB holds currently 63 max charakters for objectnames.
+        To prevent exceeds and collisions, the index names are shortend
+        based upon a hash.
+        SHA1 holds a max output of 40 characters
+        """
+        hash = hashlib.sha1()
+        hash.update(bytes(index_name, "utf-8"))
+        index_name = hash.hexdigest() + "_idx"
```

Comment:
The `update`  is not really necessary. You could easily do this in one go:

```python
hashlib.sha1(index_name.encode().hexdigest()
```

Also using `encode()` is somewhat more idiomatic.

Reply:
For what it is worth, I'm using [blake2s](https://docs.python.org/3.8/library/hashlib.html#hashlib.blake2s) in BBGA DAG. It allows you to specify a `digest_size` making it more explicit that you are after a hash of a specific size. In your code this is rather implicit in the choice of hash algorithm (`sha1`) though clearly documented in the docstring.

Anyway to generate a 40 character hash using blake2s you would do:

```python
hashlib.blake2s(index_name.encode(), digest_size=20).hexdigest()
```

Using a `digest_size` like that would obviate the need for the `assert` statement that I suggested in https://github.com/Amsterdam/schema-tools/pull/105/files#r577047793

Reply:
Verwerkt, bedankt voor deze tip